internal/bminventory/inventory.go

```diff
@@ -1668,6 +1676,113 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
+				if cluster.CPUArchitecture != common.X86CPUArchitecture && !featuresupport.IsFeatureSupported(cluster.OpenshiftVersion, models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING) {
+					err := errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed for clusters with non-x86_64 CPU architecture")
+					log.Error(err)
+					return nil, nil, common.NewApiError(http.StatusBadRequest, err)
+				}
+
+				log.Infof("Updating %s platform to baremetal platform", *cluster.Platform.Type)
+				return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+			}
+		}
+	}
+
+	log.Infof("No platform manipulation is needed. Setting platform=%v and userManagedNetworking=%v", platform, userManagedNetworking)
+	return platform, userManagedNetworking, nil
+}
+
+func getActualCreateClusterPlatformParams(log logrus.FieldLogger, platform *models.Platform, userManagedNetworking *bool, highAvailabilityMode *string) (*models.Platform, *bool, error) {
+
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		log.Error(err)
+		return nil, nil, err
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeFull {
+		if (platform == nil || *platform.Type == models.PlatformTypeBaremetal) && !swag.BoolValue(userManagedNetworking) {
+			log.Info("Cluster high-availability-mode is set to Full, setting platform type to baremetal and disable user-managed-networking")
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+
+		if swag.BoolValue(userManagedNetworking) || (platform != nil && *platform.Type == models.PlatformTypeNone) {
+			log.Info("Cluster high-availability-mode is set to Full, setting platform type to none and enable user-managed-networking")
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	} else if *highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		if platform != nil && *platform.Type == models.PlatformTypeBaremetal {
+			err := errors.Errorf("Can't set %s platform on single node OpenShift", *platform.Type)
+			log.Error(err)
+			return nil, nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+
+		if userManagedNetworking != nil && !*userManagedNetworking {
+			err := errors.New("Can't disable user-managed-networking on single node OpenShift")
+			log.Error(err)
+			return nil, nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+
+		log.Info("Cluster high-availability-mode is set to None, setting platform type to none and enable user-managed-networking")
+		return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+	}
+
+	log.Infof("No platform manipulation is needed. Setting platform=%v and userManagedNetworking=%v", platform, userManagedNetworking)
```

Comment:
This isn't really accurate because there are several cases above where the function returned the same thing that was passed.  I recommend logging the input to the function and the output.